### PR TITLE
Feature/code artifact

### PR DESCRIPTION
**Pull request by arejula27**
Implemented the codeartifect shema

### CODE REVIEW

File: src/schemas.py [hunk 1]
```diff
@@ -62,6 +62,14 @@ class AIoDPublication(AIoDAIResource):
     )
 
 
+class AIoDCodeArtifact(AIoDAIResource):
+    """The complete metadata of a code artifact. For now, only a couple of fields are shown,
+    we have to decide which fields to use."""
+
+    name: str = Field(max_length=250)
```
Comment (josvandervelde) [inline src/schemas.py:69]: These are not a lot of fields :) Should we make a start with defining some fields that we expect there to be?

**Review by josvandervelde — COMMENTED** [review]
Looks good to me!

File: src/database/model/code_artifact.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    )
+    # Required fields
+    name: Mapped[str] = mapped_column(String(250), nullable=False)
+    # Recommended fields
+    doi: Mapped[str] = mapped_column(String(150), nullable=True, default=None)
+    abstarct: Mapped[str] = mapped_column(String(5000), nullable=True, default=None)
+    hardware_resources_description: Mapped[str] = mapped_column(
+        String(5000), nullable=True, default=None
+    )
+    os_description: Mapped[str] = mapped_column(String(5000), nullable=True, default=None)
```
Comment (josvandervelde) [inline src/database/model/code_artifact.py:26]: As discussed: we should probably take some more time to enhance our datamodel. Can we fit all interesting fields from Huggingface dataset, from OpenML flow, from Zenodo Software (including Notebooks) into our definition? Do we need separate Notebook/Model/Other Code, or is a single Code Artifact better?


Moreover, can we use https://schema.org/Code? I think that `description` might be more appropriate than `abstract`, for instance.

**Comment by arejula27** [comment]
Instead of adding a list of distributions, it has been decided to put the fields of a distribution in the code artifact. This is temporary due to the urgency of adding it. In the future, this will be a separate schema so that an artifact can have multiple code distributions (such as two Docker ones).

**Comment by arejula27** [comment]
branch deprecated, another one will be created with a newr version of code artifact

**Pull request closed without merging**